src/frequenz/sdk/actor/power_distributing/_component_managers/_ev_charger_manager/_ev_charger_manager.py

```diff
@@ -334,14 +334,17 @@ async def _set_api_power(
         succeeded_components: set[int] = set()
         failed_power = Power.zero()
         for component_id, task in tasks.items():
-            exc = task.exception()
+            try:
+                exc = task.exception()
+            except asyncio.CancelledError as e:
+                exc = e
             if exc is not None:
                 failed_components.add(component_id)
                 failed_power += target_power_changes[component_id]
             else:
                 succeeded_components.add(component_id)
 
-            match task.exception():
+            match exc:
                 case asyncio.CancelledError():
                     _logger.warning(
                         "Timeout while setting power to EV charger %s", component_id
```

Comment:
A slightly simpler alternative:

```py
            try:
                task.result()
            except asyncio.CancelledError:
                _logger.warning(
                    "Timeout while setting power to EV charger %s", component_id
                )
            except asyncio.ClientError as err:
                _logger.warning(
                    "Got a client error while setting power to EV charger %s: %s",
                    component_id,
                    err,
                )
            except Exception:
                _logger.exception(
                    "Unknown error while setting power to EV charger: %s",
                    component_id,
                )
            else:
                succeeded_components.add(component_id)
                continue

            failed_components.add(component_id)
            failed_power += target_power_changes[component_id]
```

Reply:
was doing this too, should probably have mentioned. done now.